### ISSUE TITLE
go.lang.security.audit.database.string-formatted-query can not support go sql prepared statement

### ISSUE THREAD

**Issue by tuhaolam** [returntocorp/sgrep-rules#1249]
**Describe the bug**
A clear and concise description of what the bug is and for which rule. Is the rule producing a false positive? Is it incorrect or missing something?

the rule:go.lang.security.audit.database.string-formatted-query can not support go sql prepared statement, such db.Exec('select * from table where key=?', req.id), the rule will produce a false positive.

**To Reproduce**
```
184:    insertSql := "insert into t_ad_experiment (exp_layer,buckets,opposite_buckets,is_transparent, " +
185:            " description,is_full,start_time,end_time,creat_time,update_time,update_user,white_list,extra,status)" +
186:            " value (?,?,?,?,?,?,?,?,?,?,?,?,?,?)"
187:    t := time.Now().Unix()
188:    InsertResult, err := DbConn.Exec(insertSql, info.Exp_layer, info.Buckets, info.Opposite_buckets,
189:            info.Is_transparent, info.Description, info.Is_full, info.Start_time, info.End_time, t, t,
190:            session.User, info.White_list, info.Extra, 0)
```
**Expected behavior**
support prepare statement identify

**Priority**
How important is this to you?
- [x] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Additional Context**

**Comment by minusworld** [comment]
Hey @tuhaolam, thanks for filing this issue! 🙏 

The problem here is that Semgrep thinks `insertSql` in your example code is a concatenated string. If you put the entire query on one line, Semgrep understands that it is only a string literal. You can see this behavior here: https://semgrep.dev/s/NrLG

That said, this isn't ideal, so I've pushed a fix for this inconvenience and once it's merged you should be good to go. 👍 

Thanks!

**Comment by tuhaolam** [comment]
Thanks a lot. But I found that there is a other false positive, such as:
```
183:    var := user()
184:    insertSql := "insert into t_ad_experiment (exp_layer,buckets,opposite_buckets,is_transparent, " +
185:            " description,is_full,start_time,end_time,creat_time,update_time,update_user,white_list,extra,status)" +
186:            " value (?,?,?,?,?,?,?,?,?,?,?,?,?,?)" + var
187:    t := time.Now().Unix()
188:    InsertResult, err := DbConn.Exec(insertSql, info.Exp_layer, info.Buckets, info.Opposite_buckets,
189:            info.Is_transparent, info.Description, info.Is_full, info.Start_time, info.End_time, t, t,
190:            session.User, info.White_list, info.Extra, 0)
```

And then how to fix it? Thank you again:)
